### ISSUE TITLE
Specify metric via callback

### ISSUE THREAD

**Issue by lvoegtlin** [DIVA-DIA/DIVA-DAF#9]
**Is your feature request related to a problem? Please describe.**
To make the system more flexible we have to implement the metrics with callbacks s.t. we can combine multiple metrics and also reuse them in other tasks.

**Describe the solution you'd like**
Implement mIoU (jar fashion), precision, recall, and accuracy as metric callbacks.
Call metrics at the end of the steps ([see](https://torchmetrics.readthedocs.io/en/latest/pages/overview.html#metrics-in-dataparallel-dp-mode))
Also make sure that when we are testing and in ddp that we just run it on one gpu or with join ([documentation of join](https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html#torch.nn.parallel.DistributedDataParallel.join)) ([look here](https://torchmetrics.readthedocs.io/en/latest/pages/overview.html#metrics-in-distributed-data-parallel-ddp-mode) or [here](https://github.com/PyTorchLightning/pytorch-lightning/issues/3325)) 

### Checklist

- [x] Implement DIVA HisDB metric class (our metric)
- [x] Metric which is exactly like the jar
- [x] Create config for mIoU
